src/v1/users/controllers/userProfileController.js

```diff
@@ -2,10 +2,15 @@ import HttpStatus from 'http-status-codes';
 import userDao from '../dao';
 import { passwordService, modelService } from '../services';
 import jwtService from '../../../services/jwtService';
+import permissions from '../../../constants/permissions';
 
 class UserProfileController {
   constructor(DAO) {
     this.DAO = DAO;
+    this.permissionLevel = {
```

Comment:
OK, this is good. The only thing left to do is to assign it not in a constructor, but using a decorator function.

Read about JS decorators (just don't use @ syntax) and create a `permissionValidation` decorator

Reply:
Ou, It's interesting, okay.